### ISSUE TITLE
Torch.angle() with NaN gradients

### ISSUE THREAD

**Issue by minkyu-choi04** [pytorch/pytorch#66084]
## 📚 Documentation

Seeing the `torch.angle()` description (https://pytorch.org/docs/stable/generated/torch.angle.html), it says that the behavior of `torch.angle()` has been changed since 1.8.0. Following is the note from the link. 

====== Note =======
Starting in PyTorch 1.8, angle returns pi for negative real numbers, zero for non-negative real numbers, and propagates NaNs. Previously the function would return zero for all real numbers and not propagate floating-point NaNs.
================

Here, I don't understand what it means by 'propagates NaNs'. I understood it as its gradient will be NaN when the input to `Torch.angle()` is real value. However, when I tested, it seems its gradient is not Nan. 

```
    x1=torch.tensor([[0, 4, -4, 1+4j]], dtype=torch.cfloat, requires_grad=True)
    out = x1.angle()
    print(out)
    out.mean().backward()
    print(x1.grad)
```

The corresponding outputs from print are 
```
    tensor([[0.0000, 0.0000, 3.1416, 1.3258]], grad_fn=<AngleBackward>)
    tensor([[ 0.0000+0.0000j,  0.0000+0.0625j, -0.0000-0.0625j, -0.0588+0.0147j]])
```

If my understanding to the note is correct, the gradient from `angle()` when its input is real value should be Nan, but it is not. 

Could anyone help me understand when `torch.angle()` returns Nan as its gradient? Or is my understanding on the documentation is wrong?
(Code is tested in pytorch 1.9.0)

**Comment by albanD** [comment]
Hi,

We use github issues only for bugs or feature requests.
Please use the forum to ask questions: https://discuss.pytorch.org/

In this case, what it means is that if x is nan, then `x.angle()` will be nan as well (There is no backward involved).
